crates/sui-indexer-alt-graphql/src/api/types/event/filter.rs

```diff
@@ -20,11 +31,140 @@ pub(crate) struct EventFilter {
 
     /// Limit to event that occured strictly before the given checkpoint.
     pub before_checkpoint: Option<UInt53>,
-    // TODO: (henry) Implement these filters.
-    // pub sender: Option<SuiAddress>,
-    // pub transaction_digest: Option<Digest>,
-    // pub module: Option<ModuleFilter>,
-    // pub type: Option<TypeFilter>,
+
+    /// Filter on events by transaction sender address.
+    pub sender: Option<SuiAddress>,
+
+    /// Events emitted by a particular module. An event is emitted by a particular module if some function in the module is called by a PTB and emits an event.
+    ///
+    /// Modules can be filtered by their package, or package::module. We currently do not support filtering by emitting module and event type at the same time so if both are provided in one filter, the query will error.
+    pub module: Option<ModuleFilter>,
+
+    /// This field is used to specify the type of event emitted.
+    ///
+    /// Events can be filtered by their type's package, package::module, or their fully qualified type name.
+    ///
+    /// Generic types can be queried by either the generic type name, e.g. `0x2::coin::Coin`, or by the full type name, such as `0x2::coin::Coin<0x2::sui::SUI>`.
+    pub type_: Option<TypeFilter>,
+}
+
+impl EventFilter {
+    /// Builds a SQL query to select and filter events based on sender, module, and type filters.
+    /// Uses the provided transaction bounds subquery to limit results to a specific transaction range
+    pub(crate) fn query<'q>(&self, tx_bounds_subquery: Query<'q>) -> Result<Query<'q>, RpcError> {
```

Comment:
I love how simple this now is -- I did not need to think hard at all while I read this, which is how I like it.